### PR TITLE
[ST] Triage kafkaST suite, merge test focused on labels into one test. 

### PR DESCRIPTION
**Pull request by henryZrncik**
### Type of change

- Refactoring

### Description

Fixing asserts in `testLabelModificationDoesNotBreakCluster` now called `testLabelsExistenceAndManipulation`.
Merge of `testAppDomainLabels` into  `testLabelsExistenceAndManipulation`.
Merge of `testLabelsAndAnnotationForPVC ` into  `testLabelsExistenceAndManipulation`.

Usage of `KafkaPersistent` in this test instead of `kafkaEphemeral`, (as we are now testing PVC labels as well and it simplifies some waits)
Small refactors like usage of Maps for storing labels instead of 2 arrays, addition of several logger statements.  

In a nutshell, all changes are about merging 3 tests into 1,  removal of some redundant parts, modify asserts. 
`testMessagesAreStoredInDisk`  was not merged into another test as the closest test regarding functionality will be removed anyway, so for now only switch to `KafkaPersistent` took place, notes is added for now in doc. 

fixes https://github.com/strimzi/strimzi-kafka-operator/issues/8261

### CODE REVIEW

**Comment by scholzj** [comment]
You should probably give it some name describing what the Pr does.

**Review by see-quick — APPROVED** [review]
I assume you tested it locally, so the tests are working and passing. I will also trigger the regression pipeline. Thanks for the PR :)

File: systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java [hunk 1]
```diff
@@ -1081,212 +1129,193 @@ void testLabelModificationDoesNotBreakCluster(ExtensionContext extensionContext)
...
-        StrimziPodSetUtils.waitForStrimziPodSetLabelsChange(testStorage.getNamespaceName(), testStorage.getKafkaStatefulSetName(), labels);
+        LOGGER.info("---> PODS <---");
 
-        LOGGER.info("Getting labels from stateful set resource");
-        Map<String, String> kafkaLabels = StrimziPodSetUtils.getLabelsOfStrimziPodSet(testStorage.getNamespaceName(), testStorage.getKafkaStatefulSetName());
+        List<Pod> pods = kubeClient(testStorage.getNamespaceName()).listPodsByPrefixInName(testStorage.getNamespaceName(), testStorage.getClusterName());
 
-        LOGGER.info("Verifying default labels in the Kafka CR");
+        for (Pod pod : pods) {
+            LOGGER.info("Getting labels from {} pod", pod.getMetadata().getName());
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java:1139]:
```suggestion
            LOGGER.info("Verify labels of pod/{}", pod.getMetadata().getName());
```

maybe you can also print namespace

File: systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java [hunk 2]
```diff
@@ -1081,212 +1129,193 @@ void testLabelModificationDoesNotBreakCluster(ExtensionContext extensionContext)
...
         if (!Environment.isKRaftModeEnabled()) {
-            zkPods = RollingUpdateUtils.waitTillComponentHasRolled(testStorage.getNamespaceName(), testStorage.getZookeeperSelector(), 1, zkPods);
-        }
-        kafkaPods = RollingUpdateUtils.waitTillComponentHasRolled(testStorage.getNamespaceName(), testStorage.getKafkaSelector(), 3, kafkaPods);
+            Map<String, String> zooLabels = StrimziPodSetUtils.getLabelsOfStrimziPodSet(testStorage.getNamespaceName(), testStorage.getZookeeperStatefulSetName());
 
-        labels.put(labelKeys[0], labelValues[0]);
-        labels.put(labelKeys[1], labelValues[1]);
-        labels.put(labelKeys[2], labelValues[2]);
+            LOGGER.info("Getting labels from Strimzi pod set of zookeeper resource");
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java:1153]:
```suggestion
            LOGGER.info("Verify labels from StrimziPodSet of Zookeeper resource");
```

File: systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java [hunk 3]
```diff
@@ -1081,212 +1129,193 @@ void testLabelModificationDoesNotBreakCluster(ExtensionContext extensionContext)
...
 
-        LOGGER.info("Verifying via kafka pods");
-        labels = kubeClient(testStorage.getNamespaceName()).getPod(testStorage.getNamespaceName(), KafkaResources.kafkaPodName(testStorage.getClusterName(), 0)).getMetadata().getLabels();
+        List<ConfigMap> configMaps = kubeClient(testStorage.getNamespaceName()).listConfigMapsInSpecificNamespace(testStorage.getNamespaceName(), testStorage.getClusterName());
 
-        assertThat("Label exists in kafka pods", labelValues[0].equals(labels.get(labelKeys[0])));
-        assertThat("Label exists in kafka pods", labelValues[1].equals(labels.get(labelKeys[1])));
-        assertThat("Label exists in kafka pods", labelValues[2].equals(labels.get(labelKeys[2])));
+        for (ConfigMap configMap : configMaps) {
+            LOGGER.info("Getting labels from {} config map", configMap.getMetadata().getName());
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java:1184]:
```suggestion
            LOGGER.info("Verify labels from ConfigMap/{}", configMap.getMetadata().getName());
```

File: systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java [hunk 4]
```diff
@@ -1081,212 +1129,193 @@ void testLabelModificationDoesNotBreakCluster(ExtensionContext extensionContext)
...
-            resource.getMetadata().getLabels().remove(labelKeys[1]);
-            resource.getMetadata().getLabels().remove(labelKeys[2]);
-        }, testStorage.getNamespaceName());
+        List<PersistentVolumeClaim> pvcs = kubeClient(testStorage.getNamespaceName()).listPersistentVolumeClaims(testStorage.getNamespaceName(), testStorage.getClusterName()).stream().filter(
+            persistentVolumeClaim -> persistentVolumeClaim.getMetadata().getName().contains(testStorage.getClusterName())).collect(Collectors.toList());
 
-        if (!Environment.isKRaftModeEnabled()) {
-            RollingUpdateUtils.waitTillComponentHasRolled(testStorage.getNamespaceName(), testStorage.getZookeeperSelector(), 1, zkPods);
+        for (PersistentVolumeClaim pvc : pvcs) {
+            LOGGER.info("Getting labels from {} pvc", pvc.getMetadata().getName());
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java:1194]:
```suggestion
            LOGGER.info("Verify labels from {} pvc", pvc.getMetadata().getName());
```

File: systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java [hunk 5]
```diff
@@ -1081,212 +1129,193 @@ void testLabelModificationDoesNotBreakCluster(ExtensionContext extensionContext)
...
-        LOGGER.info("Waiting for kafka service labels deletion {}", labels.toString());
-        ServiceUtils.waitForServiceLabelsDeletion(testStorage.getNamespaceName(), KafkaResources.brokersServiceName(testStorage.getClusterName()), labelKeys[0], labelKeys[1], labelKeys[2]);
+        LOGGER.info("---> Test Customer specified labels <--");
 
-        LOGGER.info("Verifying kafka labels via services");
-        service = kubeClient(testStorage.getNamespaceName()).getService(testStorage.getNamespaceName(), KafkaResources.brokersServiceName(testStorage.getClusterName()));
+        LOGGER.info("---> STRIMZI POD SETS <---");
 
-        verifyNullLabels(labelKeys, service);
+        LOGGER.info("Waiting for kafka Strimzi pod set labels existence {}", customSpecifiedLabels);
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java:1202]:
```suggestion
        LOGGER.info("Waiting for Kafka StrimziPodSet labels existence {}", customSpecifiedLabels);
```

maybe add podset name and namespace?

File: systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java [hunk 6]
```diff
@@ -1081,212 +1129,193 @@ void testLabelModificationDoesNotBreakCluster(ExtensionContext extensionContext)
...
-            LOGGER.info("Waiting for Kafka ConfigMap {}/{} to have labels removed: {}", testStorage.getNamespaceName(), cmName, labelKeys);
-            ConfigMapUtils.waitForConfigMapLabelsDeletion(testStorage.getNamespaceName(), cmName, labelKeys[0], labelKeys[1], labelKeys[2]);
-
-            LOGGER.info("Verifying Kafka labels on ConfigMap {}/{}", testStorage.getNamespaceName(), cmName);
-            ConfigMap configMap = kubeClient(testStorage.getNamespaceName()).getConfigMap(testStorage.getNamespaceName(), cmName);
+        LOGGER.info("Getting labels from strimzi pod set set resource");
+        kafkaLabelsObtained = StrimziPodSetUtils.getLabelsOfStrimziPodSet(testStorage.getNamespaceName(), testStorage.getKafkaStatefulSetName());
 
-            verifyNullLabels(labelKeys, configMap);
+        LOGGER.info("Asserting presence of custom labels which should be available in kafka with labels {}", kafkaLabelsObtained);
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java:1208]:
```suggestion
        LOGGER.info("Asserting presence of custom labels which should be available in Kafka with labels {}", kafkaLabelsObtained);
```

File: systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java [hunk 7]
```diff
@@ -1081,212 +1129,193 @@ void testLabelModificationDoesNotBreakCluster(ExtensionContext extensionContext)
...
-        LOGGER.info("Removing labels: {} -> {}, {} -> {}, {} -> {}", labelKeys[0], labels.get(labelKeys[0]),
-            labelKeys[1], labels.get(labelKeys[1]), labelKeys[2], labels.get(labelKeys[2]));
+        LOGGER.info("---> PVC (both labels and annotation) <---");
 
-        KafkaResource.replaceKafkaResourceInSpecificNamespace(testStorage.getClusterName(), resource -> {
-            resource.getMetadata().getLabels().remove(labelKeys[0]);
-            resource.getMetadata().getLabels().remove(labelKeys[1]);
-            resource.getMetadata().getLabels().remove(labelKeys[2]);
-        }, testStorage.getNamespaceName());
+        List<PersistentVolumeClaim> pvcs = kubeClient(testStorage.getNamespaceName()).listPersistentVolumeClaims(testStorage.getNamespaceName(), testStorage.getClusterName()).stream().filter(
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java:1190]:
```suggestion
        List<PersistentVolumeClaim> pvcs = kubeClient().listPersistentVolumeClaims(testStorage.getNamespaceName(), testStorage.getClusterName()).stream().filter(
```
also, when you are listing the PVCs this way, you don't need to filter it afterwards again or?

File: systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java [hunk 8]
```diff
@@ -1081,212 +1129,193 @@ void testLabelModificationDoesNotBreakCluster(ExtensionContext extensionContext)
...
+        LOGGER.info("---> PVC (both labels and annotation) <---");
 
-        Map<String, String> kafkaLabels = StrimziPodSetUtils.getLabelsOfStrimziPodSet(testStorage.getNamespaceName(), testStorage.getKafkaStatefulSetName());
+        LOGGER.info("Wait for changes in PVC labels and until kafka becomes ready");
+        PersistentVolumeClaimUtils.waitUntilPVCLabelsChange(testStorage.getNamespaceName(), testStorage.getClusterName(), customSpecifiedLabelOrAnnotationPvc, pvcLabelOrAnnotationKey);
+        PersistentVolumeClaimUtils.waitUntilPVCAnnotationChange(testStorage.getNamespaceName(), testStorage.getClusterName(), customSpecifiedLabelOrAnnotationPvc, pvcLabelOrAnnotationKey);
 
-        LOGGER.info("Getting labels from stateful set of kafka resource");
-        verifyAppLabels(kafkaLabels);
+        pvcs = kubeClient(testStorage.getNamespaceName()).listPersistentVolumeClaims(testStorage.getNamespaceName(), testStorage.getClusterName()).stream().filter(
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java:1270]:
```suggestion
        pvcs = kubeClient().listPersistentVolumeClaims(testStorage.getNamespaceName(), testStorage.getClusterName()).stream().filter(
```

+ plus same comment as above

File: systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java [hunk 9]
```diff
@@ -1081,212 +1129,193 @@ void testLabelModificationDoesNotBreakCluster(ExtensionContext extensionContext)
...
-                .collect(Collectors.toList());
+        LOGGER.info("Waiting for kafka service labels changed {}", customSpecifiedLabels);
+        ServiceUtils.waitForServiceLabelsChange(testStorage.getNamespaceName(), KafkaResources.brokersServiceName(testStorage.getClusterName()), customSpecifiedLabels);
 
-        for (Service service : services) {
-            LOGGER.info("Getting labels from {} service", service.getMetadata().getName());
-            verifyAppLabels(service.getMetadata().getLabels());
-        }
+        LOGGER.info("Verifying kafka labels via services");
+        Service service = kubeClient(testStorage.getNamespaceName()).getService(testStorage.getNamespaceName(), KafkaResources.brokersServiceName(testStorage.getClusterName()));
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java:1287]:
```suggestion
        Service service = kubeClient().getService(testStorage.getNamespaceName(), KafkaResources.brokersServiceName(testStorage.getClusterName()));
```

File: systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java [hunk 10]
```diff
@@ -1081,212 +1129,193 @@ void testLabelModificationDoesNotBreakCluster(ExtensionContext extensionContext)
...
+        StrimziPodSetUtils.waitForStrimziPodSetLabelsChange(testStorage.getNamespaceName(), testStorage.getKafkaStatefulSetName(), customSpecifiedLabels);
 
-        for (ConfigMap configMap : configMaps) {
-            LOGGER.info("Getting labels from {} config map", configMap.getMetadata().getName());
-            verifyAppLabelsForSecretsAndConfigMaps(configMap.getMetadata().getLabels());
+        LOGGER.info("Verifying kafka labels via strimzi pod set");
+        verifyPresentLabels(customSpecifiedLabels, StrimziPodSetUtils.getLabelsOfStrimziPodSet(testStorage.getNamespaceName(), testStorage.getKafkaStatefulSetName()));
+
+        LOGGER.info("Verifying via kafka pods");
+        Map<String, String> podLabels = kubeClient(testStorage.getNamespaceName()).getPod(testStorage.getNamespaceName(), KafkaResources.kafkaPodName(testStorage.getClusterName(), 0)).getMetadata().getLabels();
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java:1312]:
```suggestion
        Map<String, String> podLabels = kubeClient().getPod(testStorage.getNamespaceName(), KafkaResources.kafkaPodName(testStorage.getClusterName(), 0)).getMetadata().getLabels();
```

File: systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java [hunk 11]
```diff
@@ -1735,13 +1644,13 @@ void verifyPresentLabels(Map<String, String> labels, Map<String, String> resourc
         }
     }
 
-    void verifyNullLabels(String[] labelKeys, Map<String, String> labels) {
+    void verifyNullLabels(Set<String> labelKeys, Map<String, String> labels) {
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java:1647]: I don't see it in the code, but you can maybe use List and go through the forEach ? It's just a nit, but when you are changing it 😄 (same for the second method).

But if it would mean that you will have to change other tests, just leave it for now as it is.

File: systemtest/src/main/java/io/strimzi/systemtest/utils/kubeUtils/controllers/ConfigMapUtils.java [hunk 12]
```diff
@@ -48,7 +49,7 @@ public static void waitForConfigMapLabelsChange(String namespaceName, String con
         }
     }
 
-    public static void waitForConfigMapLabelsDeletion(String namespaceName, String configMapName, String... labelKeys) {
+    public static void waitForConfigMapLabelsDeletion(String namespaceName, String configMapName, Collection<String> labelKeys) {
```
Comment (im-konge) [inline systemtest/src/main/java/io/strimzi/systemtest/utils/kubeUtils/controllers/ConfigMapUtils.java:52]: Why is this needed? I can't see the benefit - you are not using anything from the `Collection` - or is there something I don't see?

**Review by im-konge — COMMENTED** [review]
Good job :) just few nits.

File: systemtest/src/main/java/io/strimzi/systemtest/utils/kubeUtils/controllers/ConfigMapUtils.java (hunk 12 above)
Comment (henryZrncik) [reply systemtest/src/main/java/io/strimzi/systemtest/utils/kubeUtils/controllers/ConfigMapUtils.java:52]: I removed this method same as the one referenced above.

File: systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java (hunk 11 above)
Comment (henryZrncik) [reply systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java:1647]: solution same as in comment lower is the same (I did not notice that this method is used only in part of code i actually deleted as part of this issue), thus I actually removed it as well now.

File: systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java [hunk 13]
```diff
@@ -1081,212 +1129,193 @@ void testLabelModificationDoesNotBreakCluster(ExtensionContext extensionContext)
...
-                "name-of-the-label-1", labelValues[0], "name-of-the-label-2", labelValues[1], labelKeys[2], labelValues[2]);
+        Map<String, String> kafkaLabelsObtained = StrimziPodSetUtils.getLabelsOfStrimziPodSet(testStorage.getNamespaceName(), testStorage.getKafkaStatefulSetName());
 
-        LOGGER.info("Edit kafka labels in Kafka CR");
-        KafkaResource.replaceKafkaResourceInSpecificNamespace(testStorage.getClusterName(), resource -> {
-            resource.getMetadata().getLabels().put(labelKeys[0], labelValues[0]);
-            resource.getMetadata().getLabels().put(labelKeys[1], labelValues[1]);
-            resource.getMetadata().getLabels().put(labelKeys[2], labelValues[2]);
-        }, testStorage.getNamespaceName());
+        LOGGER.info("Getting labels from Strimzi pod set of kafka resource");
```
[the remark this answers is not in the document]
Comment (henryZrncik) [reply systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java]: thank you, resolved as all others LOGGER related suggestions.

File: systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java [hunk 14]
```diff
@@ -1081,212 +1129,193 @@ void testLabelModificationDoesNotBreakCluster(ExtensionContext extensionContext)
...
-        StrimziPodSetUtils.waitForStrimziPodSetLabelsChange(testStorage.getNamespaceName(), testStorage.getKafkaStatefulSetName(), labels);
-
-        LOGGER.info("Verifying kafka labels via stateful set");
-
-        verifyPresentLabels(labels, StrimziPodSetUtils.getLabelsOfStrimziPodSet(testStorage.getNamespaceName(), testStorage.getKafkaStatefulSetName()));
+        LOGGER.info("---> CONFIG MAPS <---");
 
-        LOGGER.info("Verifying via kafka pods");
-        labels = kubeClient(testStorage.getNamespaceName()).getPod(testStorage.getNamespaceName(), KafkaResources.kafkaPodName(testStorage.getClusterName(), 0)).getMetadata().getLabels();
+        List<ConfigMap> configMaps = kubeClient(testStorage.getNamespaceName()).listConfigMapsInSpecificNamespace(testStorage.getNamespaceName(), testStorage.getClusterName());
```
[the remark this answers is not in the document]
Comment (henryZrncik) [reply systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java]: thanks Lukas, fixed!  redundant argument is no longer provided. (same below)

File: systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java [hunk 15]
```diff
@@ -1050,24 +978,90 @@ void testRegenerateCertExternalAddressChange(ExtensionContext extensionContext)
...
+     *     - Producer and Consumer are able to produce and consume messages respectively.
+     *  3. - Modify configuration of Kafka CR with addition of new labels and modification of existing
+     *     - Kafka is rolling and new labels are present in Kafka CR, and managed resources
+     *  4. - Deploy Producer and Consumer configured to produce and consume default number of messages, to make sure Kafka works as expected
+     *     - Producer and Consumer are able to produce and consume messages respectively.
+     *
+     * @usecase
+     *  - annotations
+     *  - labels
+     */
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java:999]:
```suggestion
    /**
     * @description This test case verifies the presence of expected Strimzi specific labels, also labels and annotations specified by user.
     * Some of user-specified labels are later modified (new is added, one is modified) which triggers rolling update after which
     * all changes took place as expected.
     *
     * @steps
     *  1. - Deploy Kafka with persistent storage and specify custom labels in CR metadata, and also other labels and annotation in PVC metadata
     *     - Kafka is deployed with its default labels and all others specified by user.
     *  2. - Deploy Producer and Consumer configured to produce and consume default number of messages, to make sure Kafka works as expected
     *     - Producer and Consumer are able to produce and consume messages respectively.
     *  3. - Modify configuration of Kafka CR with addition of new labels and modification of existing
     *     - Kafka is rolling and new labels are present in Kafka CR, and managed resources
     *  4. - Deploy Producer and Consumer configured to produce and consume default number of messages, to make sure Kafka works as expected
     *     - Producer and Consumer are able to produce and consume messages respectively.
     *
     * @usecase
     *  - annotations
     *  - labels
     *  - kafka-rolling-update
     *  - persistent-storage
     */
```

File: systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java [hunk 16]
```diff
@@ -1081,212 +1075,193 @@ void testLabelModificationDoesNotBreakCluster(ExtensionContext extensionContext)
...
+            resource.getSpec().getKafka().getTemplate().getPersistentVolumeClaim().getMetadata().setAnnotations(customSpecifiedLabelOrAnnotationPvc);
+            resource.getSpec().getZookeeper().getTemplate().getPersistentVolumeClaim().getMetadata().setLabels(customSpecifiedLabelOrAnnotationPvc);
+            resource.getSpec().getZookeeper().getTemplate().getPersistentVolumeClaim().getMetadata().setAnnotations(customSpecifiedLabelOrAnnotationPvc);
+        }, testStorage.getNamespaceName());
 
-        for (Pod pod : pods) {
-            LOGGER.info("Getting labels from {} pod", pod.getMetadata().getName());
-            verifyAppLabels(pod.getMetadata().getLabels());
-        }
+        LOGGER.info("Wait for rollout of Zookeeper and Kafka");
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java:1206]: for rolling update instead of rollout?

**Review by Frawless — APPROVED** [review]
Just a nits

File: systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java (hunk 15 above)
Comment (henryZrncik) [reply systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java]: thanks for corrections. 👍

File: systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java (hunk 16 above)
Comment (henryZrncik) [reply systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java]: 👍 changed. both works in my case but your suggestion is more specific which is good.

File: systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java [hunk 17]
```diff
@@ -1050,24 +978,92 @@ void testRegenerateCertExternalAddressChange(ExtensionContext extensionContext)
         });
     }
 
+    /**
+     * @description This test case verifies the presence of expected Strimzi specific labels, also labels and annotations specified by user.
+     * Some of user-specified labels are later modified (new is added, one is modified) which triggers rolling update after which
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java:983]:
```suggestion
     * Some of user-specified labels are later modified (new are added, one is modified) which triggers rolling update after which
```
or
```suggestion
     * Some of user-specified labels are later modified (new one is added, one is modified) which triggers rolling update after which
```

**Review by im-konge — APPROVED** [review]
One last suggestion, LGTM otherwise.

Good job :)

File: systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java (hunk 17 above)
Comment (henryZrncik) [reply systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java]: 👌

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=regression testcase=KafkaST

**Comment by Frawless** [comment]
Jenkins failure is connected to infra - the failed test uses loadbalancer which is not available on it. All other tests passed so we should be fine with merging this.
